pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -564,7 +564,13 @@ func NewGarbageCollector(metaOnlyClientPool dynamic.ClientPool, clientPool dynam
 		}
 		kind, err := gc.restMapper.KindFor(resource)
 		if err != nil {
-			return nil, err
+			if _, ok := err.(*meta.NoResourceMatchError); ok {
+				// NoResourceMatchError would be returned for TPR
+				glog.Warningf("ignore NoResourceMatchError for %v", resource)
+				continue
+			} else {
```

Comment:
@sttts don't we need to return the error when it is not of type NoResourceMatchError?